tchannel-core/src/main/java/com/uber/tchannel/handlers/InitRequestHandler.java

```diff
@@ -64,6 +76,7 @@ public void messageReceived(ChannelHandlerContext ctx, Message message) throws E
 
             default:
 
+                // TODO: should send back BadRequest
```

Comment:
yes, should be ProtocolError. Exception should not be thrown here. TChannel should just send back protocol error and reset the connection. All for future PRs ...